Sources/PubNub/Helpers/WeakBox.swift

```diff
@@ -29,43 +29,51 @@ final class WeakBox<Element>: Hashable where Element: AnyObject, Element: Hashab
 }
 
 struct WeakSet<Element> where Element: AnyObject, Element: Hashable {
-  private var elements: Set<WeakBox<Element>> = []
+  private var elements: Atomic<Set<WeakBox<Element>>> = Atomic([])
 
   init(_ elements: [Element]) {
-    elements.forEach { self.elements.update(with: WeakBox($0)) }
+    self.elements.lockedWrite { [elements] currentValue in
+      elements.forEach { element in
+        currentValue.update(with: WeakBox(element))
+      }
+    }
   }
 
   // NSSet Operations
   var allObjects: [Element] {
-    return elements.compactMap { $0.underlying }
+    return elements.lockedRead { $0 }.compactMap { $0.underlying }
```

Comment:
`compactMap` after closure, close will operate on the reference to the original `elements` (they did not mutate in `lockedRead` closure and COW not applied) but outside of lock. The following change should be more thread-safe:
```suggestion
    return elements.lockedRead { $0.compactMap { $0.underlying } }
```